crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -642,3 +648,67 @@ async fn report_content() {
 
     room.report_content(event_id, Some(score), Some(reason.to_owned())).await.unwrap();
 }
+
+#[async_test]
+async fn subscribe_to_typing_notifications() {
+    let (client, server) = logged_in_client().await;
+    let typing_sequences: Arc<Mutex<Vec<Vec<OwnedUserId>>>> = Arc::new(Mutex::new(Vec::new()));
+    let asserted_typing_sequences =
+        vec![vec![user_id!("@alice:matrix.org"), user_id!("@bob:example.com")], vec![]];
+    let room_id = room_id!("!test:example.org");
+    let mut ev_builder = SyncResponseBuilder::new();
+
+    // Initial sync with our test room.
+    ev_builder.add_joined_room(JoinedRoomBuilder::new(room_id));
+    mock_sync(&server, ev_builder.build_json_sync_response(), None).await;
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+    let _response = client.sync_once(sync_settings.clone()).await.unwrap();
+    server.reset().await;
+
+    // Send to typing notification
+    let room = client.get_room(room_id).unwrap();
+    let typing_seq = Arc::clone(&typing_sequences);
+    let handle = tokio::spawn(async move {
+        let (_guard, mut subscriber) = room.subscribe_to_typing_notifications();
+        while let Ok(typing_users) = subscriber.recv().await {
+            let mut typings = typing_seq.lock().unwrap();
+            typings.push(typing_users);
+            if typings.len() == 2 {
+                break;
+            }
+        }
+    });
```

Comment:
You can move `typing_seq` inside a new block, that's just simpler and avoid creating a new variable name which is confusing. Same inside the `while` block, let's use the same name. Also, you can use `.clone()` directly instead of `Arc::clone`.

```suggestion
    let handle = tokio::spawn({
        let typing_sequences = typing_sequences.clone();

        async move {
            let (_guard, mut subscriber) = room.subscribe_to_typing_notifications();

            while let Ok(typing_users) = subscriber.recv().await {
                let mut typing_sequences = typing_sequences.lock().unwrap();
                typing_sequences.push(typing_users);

                if typing_sequences.len() == 2 {
                    break;
                }
            }
        }
    });
```

Feel free to add empty lines here and there to make the code easier to read please.